x/globalfee/ante/fee.go

```diff
@@ -0,0 +1,347 @@
+package ante
+
+import (
+	"encoding/json"
+	"errors"
+
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/public-awesome/stargaze/v9/x/globalfee/types"
+)
+
+var _ sdk.AnteDecorator = FeeDecorator{}
+
+type GlobalFeeReaderExpected interface {
+	GetContractAuthorization(ctx sdk.Context, contractAddr sdk.AccAddress) (types.ContractAuthorization, bool)
+	GetCodeAuthorization(ctx sdk.Context, codeId uint64) (types.CodeAuthorization, bool)
+	GetContractInfo(ctx sdk.Context, contractAddress sdk.AccAddress) *wasmtypes.ContractInfo
+	GetParams(ctx sdk.Context) types.Params
+}
+
+type StakingReaderExpected interface {
+	BondDenom(ctx sdk.Context) string
+}
+
+type FeeDecorator struct {
+	codec         codec.BinaryCodec
+	feeKeeper     GlobalFeeReaderExpected
+	stakingKeeper StakingReaderExpected
+}
+
+func NewFeeDecorator(codec codec.BinaryCodec, fk GlobalFeeReaderExpected, sk StakingReaderExpected) FeeDecorator {
+	return FeeDecorator{
+		codec:         codec,
+		feeKeeper:     fk,
+		stakingKeeper: sk,
+	}
+}
+
+// AnteHandle implements the AnteDecorator interface
+func (mfd FeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must implement the sdk.FeeTx interface")
+	}
+
+	// Only check for minimum fees and global fee if the execution mode is CheckTx
+	if !ctx.IsCheckTx() || simulate {
+		return next(ctx, tx, simulate)
+	}
```

Comment:
With this validators can still propose 0fee txs right ?

Reply:
yes, in case when the validator has configured `minimum-gas-prices = "0ustars"`